### ISSUE TITLE
Bad link provided when DTL is opened from IRIS for Health 2023.2

### ISSUE THREAD

**Issue by gjsjohnmurray** [intersystems-community/vscode-objectscript#1228]
The extension submits this REST GET request:
```
/api/atelier/v1/HSCUSTOM/doc/Ens.SSH.InteractiveAuth.DTL.dtl
```
and receives this response:
```json
{
  "status": {
    "errors": [],
    "summary": ""
  },
  "console": [],
  "result": {
    "name": "Ens.SSH.InteractiveAuth.DTL.dtl",
    "db": "@OTHER",
    "ts": "2023-07-31 17:14:19.000",
    "upd": true,
    "cat": "OTH",
    "status": "",
    "enc": false,
    "flags": 0,
    "content": [
      "/csp/hscustom/EnsPortal.DTLEditor.zen?DT=Ens.SSH.InteractiveAuth.DTL.DTL&STUDIO=1"
    ]
  }
}
```

It strips the &STUDIO=1 queryparam from the URL in "content", then offers the VS Code user a document containing the link.

However on my target server the correct URL ends thus:
```
/csp/healthshare/hscustom/EnsPortal.DTLEditor.zen?DT=Ens.SSH.InteractiveAuth.DTL.DTL
```

Notice the extra `/healthshare' segment in the path.

```
HSCUSTOM>w $zv
IRIS for UNIX (Ubuntu Server 22.04 LTS for x86-64) 2023.2 (Build 227U) Mon Jul 31 2023 18:09:44 EDT
HSCUSTOM>

```

Ideally this needs to be fixed server-side, but meanwhile his code could fix up the URL as long as we know what the rules are (i.e. what server version and variant requires the additional `/healthshare` segment).

https://github.com/intersystems-community/vscode-objectscript/blob/08c18a0644967d6bd8229065538576a1f90ae90f/src/providers/FileSystemProvider/FileSystemProvider.ts#L725-L734

The same fixup will always be needed here anyway for when the user creates a new DTL file:

https://github.com/intersystems-community/vscode-objectscript/blob/08c18a0644967d6bd8229065538576a1f90ae90f/src/commands/newFile.ts#L649-L653

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray I will look into this internally

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray If you click on the link shown in VS Code, does it take you to the editor?

**Comment by gjsjohnmurray** [comment]
No, it gives a "Not found" page. If I edit the URL to add the `/healthshare' segment it succeeds.

**Comment by isc-bsaviano** [comment]
Is the `$ZV` you posted before correct? I don't see an IRIS for Health suffix

**Comment by gjsjohnmurray** [comment]
I have just re-validated by doing a fresh install from IRISHealth_Community-2023.2.0.227.0-win_x64.exe downloaded from online distribution page. The problem occurs there too, and the $ZV I get from this is:
```
HSCUSTOM>w $zv
IRIS for Windows (x86-64) 2023.2 (Build 227U) Mon Jul 31 2023 17:48:09 EDT
HSCUSTOM>
```

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray Maybe this is something specific to healthshare namespaces? The link without the healthshare part works if you use the USER namespace.

**Comment by isc-bsaviano** [comment]
I can ask around and see if anyone internally knows why this would be happening.

**Comment by gjsjohnmurray** [comment]
There's a server-side classmethod to discover whether a namespace is HealthShare-enabled:

https://docs.intersystems.com/irislatest/csp/documatic/%25CSP.Documatic.cls?LIBRARY=%25SYS&CLASSNAME=%25Library.EnsembleMgr#IsHealthShareNamespace

Don't know how we can use that over /api/atelier to discover if the current namespace is HealthShare-enabled though.

**Comment by isc-bsaviano** [comment]
This isn't an Atelier API related bug. The URL is wrong in the source that Atelier gets it from.

**Comment by gjsjohnmurray** [comment]
Looks like the HSCUSTOM is reported as **not** a HealthShare namespace:

```
%SYS>w ##class(%EnsembleMgr).IsHealthShareNamespace("HSCUSTOM")
0
%SYS>
```

but nor does it get given a `/csp/hscustom` web app definition.

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray I reported this to the correct developer internally

**Comment by isc-bsaviano** [comment]
@gjsjohnmurray This should be fixed in the first preview kit for 2023.3
